mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -1422,3 +1424,261 @@ def _auto_indent(repr_str, indent):
+
+        return results
+
+    def _affine_bboxes(self, results, trans_affine):
+        """Affine transform bboxes to input image.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+            trans_affine (np.ndarray): Matrix transforming original
+                image to the network input image size.
+        """
+
+        for key in results.get('bbox_fields', []):
+            bboxes = results[key]
+            bboxes[:, :2] = self._affine_transform(bboxes[:, :2], trans_affine)
+            bboxes[:, 2:] = self._affine_transform(bboxes[:, 2:], trans_affine)
+            if self.bbox_clip_border:
+                bboxes[:,
+                       [0, 2]] = bboxes[:,
+                                        [0, 2]].clip(0, self.img_scale[0] - 1)
+                bboxes[:,
+                       [1, 3]] = bboxes[:,
+                                        [1, 3]].clip(0, self.img_scale[1] - 1)
+            results[key] = bboxes
+
+    def _affine_transform(self, point, matrix):
+        """Affine transform bbox points to input iamge.
+
+        Args:
+            points (np.ndarray): Points to be transformed.
+                shape: (N, 2)
+            matrix (np.ndarray): Affine transform matrix.
+                shape: (3, 3)
+
+        Returns:
+            np.ndarray: Transformed points.
+        """
+        point_num = point.shape[0]
+        point_exd = np.concatenate((point, np.ones((point_num, 1))), axis=1)
+        point_exd_trans = point_exd.T
+        new_point = np.matmul(matrix, point_exd_trans).T
+        return new_point[:, :2]
+
+    def _get_transfrom_matrix(self, center_scale, output_size):
+        """Get affine transform matrix.
+
+        Args:
+            center_scale (list[tuple]): Center and scale of
+                current image.
+            output_size (tuple): The transform target size.
+
+        Returns:
+            np.ndarray: Affine transform matrix.
+        """
+        center, scale = center_scale[0], center_scale[1]
+        # todo: further add rot and shift here.
+        src_w = scale[0]
+        dst_w = output_size[0]
+        dst_h = output_size[1]
+
+        src_dir = np.array([0, src_w * -0.5])
+        dst_dir = np.array([0, dst_w * -0.5])
+
+        src = np.zeros((3, 2), dtype=np.float32)
+        dst = np.zeros((3, 2), dtype=np.float32)
+        src[0, :] = center
+        src[1, :] = center + src_dir
+        dst[0, :] = np.array([dst_w * 0.5, dst_h * 0.5])
+        dst[1, :] = np.array([dst_w * 0.5, dst_h * 0.5]) + dst_dir
+
+        src[2, :] = self._get_3rd_point(src[0, :], src[1, :])
+        dst[2, :] = self._get_3rd_point(dst[0, :], dst[1, :])
+
+        get_matrix = cv2.getAffineTransform(src, dst)
+
+        matrix = np.concatenate((get_matrix, [[0., 0., 1.]]))
+
+        return matrix.astype(np.float32)
+
+    def _get_3rd_point(self, point_a, point_b):
+        """Get 3rd point to calculate affine transfrom matrix."""
```

Comment:
3rd point is a little unclear? maybe use ref_point or sth.? Or this term is called like this in the original paper, then add a comment/reference here.

Reply:
I'm not sure the point_a and point_b can be good variable names, maybe to unify these two variable with ref_point, we can name them ref_point1, ref_point2, ref_point3??